### ISSUE TITLE
debbuild fails with `can't exec:`

### ISSUE THREAD

**Issue by Conan-Kudo** [ascherer/debbuild#61]
When testing debbuild 17.2.0 on build.opensuse.org with a GNU Hello spec, debbuild failed to run with the following output:

```
This is debbuild, Version 17.2.0
Can't exec: 
```

See: https://build.opensuse.org/package/show/home:Pharaoh_Atem/hello

Example log: https://build.opensuse.org/build/home:Pharaoh_Atem/xUbuntu_16.04/x86_64/hello/_log

SRPM of test package: http://download.opensuse.org/repositories/home:/Pharaoh_Atem/CentOS_7/src/hello-2.10-61.1.el7.src.rpm

Spec of debbuild package: https://build.opensuse.org/package/view_file/home:Pharaoh_Atem/debbuild/debbuild.spec?rev=27

I've been able to reproduce it locally on Debian 8 and Ubuntu 16.04.

**Comment by ascherer** [comment]
In your specfile, please replace line 52
```
%setup -q -n %{name}-%{name}-%{version}
```
with the caconical
```
%setup -q -n %{name}-%{version}
```
and all will be fine.

**Comment by Conan-Kudo** [comment]
@ascherer If I do that, the `debbuild` package build fails. debbuild built fine on older debbuild. The issue is that new debbuild can't build GNU Hello. I probably can't build new debbuild on new debbuild either.

**Comment by ascherer** [comment]
OK, I'll test with the download archive. However, with my local `git archive -o debbuild-17.2.0.tar.gz --prefix debbuild-17.2.0/ debbuild-17.2.0` and the single modification to your rebuild.spec, I get a clean build (apart from a few "virtual package" warnings).

**Comment by Conan-Kudo** [comment]
@ascherer I use the URL generated by GitHub, since you don't upload an official tarball: https://github.com/ascherer/debbuild/archive/debbuild-17.2.0.tar.gz

**Comment by ascherer** [comment]
I'll fix the messy download files RSN.

If you run `debbuild -ba -vv SPECS/debbuild.spec`, you'll notice that `%setup` looks for `SOURCES/debbuild-17.2.0.tar.gz`. With a suitable softlink to `debbuild-debbuild-17.2.0.tar.gz`, it 
works.

Your `hello` drop hickups on `%make_build`. I will investigate ASAP.

**Comment by ascherer** [comment]
1. Messy download files: Github combines the **project name** and the **tag** when creating the source code files and the internal prefix, although, when _hovering_ over **Source code (.tar.gz)**, you'll only see **tag.tar.gz**. I changed all tags from `debbuild-17.2.1` to `17.2.1`.
2. Building **debbuild**: Fixed with the new download file(s) and the modified **debbuild.spec** as above. I'd suggest to download **debbuild-17.2.1-ascherer.ubuntu16.04.sdeb** and run `debbuild -i debbuild-17.2.1-ascherer.ubuntu16.04.sdeb` in your `%_topdir`. This places both the _correct_ tarball and the specfile in the right places for the following `debbuild -ba`.
3. Building **hello**, resp., incorrect expansion of `%make_build`: That was indeed a bug in `debbuild`. Such simple `%macros` were _not_ expanded in multi-line scripts. Fixed in version **17.2.1**.

**Comment by Conan-Kudo** [comment]
In re download files, it means that https://github.com/ascherer/debbuild/archive/17.2.1/debbuild-17.2.1.tar.gz will give you the correct, expected tarball.

**Comment by ascherer** [comment]
Note that **debbuild 17.2.0** and earlier _requires_ to be called with the options `-i -v` to work correctly. This has been fixed in **debbuild 17.2.1** as well.

**Comment by Conan-Kudo** [comment]
Looks fixed to me!
